### PR TITLE
subscribeEvent() to use JS promise

### PR DESCRIPTION
**Pull request by fillobotto**
- [x] Coap
- [x] Modbus
- [x] MQTT
- [x] HTTP
- [x] Opcua

### CODE REVIEW

File: packages/binding-coap/src/coap-client.ts [hunk 1]
```diff
@@ -140,29 +140,33 @@ export default class CoapClient implements ProtocolClient {
     });
   }
 
-  public subscribeResource(form: CoapForm, next: ((value: any) => void), error?: (error: any) => void, complete?: () => void): any {
-    let req = this.generateRequest(form, "GET", true);
+  public subscribeResource(form: CoapForm, next: ((value: any) => void), error?: (error: any) => void, complete?: () => void): Promise<Subscription> {
+    return new Promise<Subscription>((resolve, reject) => {
+      let req = this.generateRequest(form, "GET", true);
```
Comment (relu91) [inline packages/binding-coap/src/coap-client.ts]: Here we are doing a get not an observe... how is it working? on coaps we are correctly issuing an observe request.

File: packages/binding-modbus/src/modbus-client.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { SecurityScheme } from '@node-wot/td-tools'
 import { modbusFunctionToEntity } from './utils'
 import { ModbusConnection, PropertyOperation } from './modbus-connection'
 import { Readable } from 'stream'
+import { Subscription } from "rxjs/Subscription"
```
Comment (relu91) [inline packages/binding-modbus/src/modbus-client.ts]: Shouldn't be already provided by the core module? how did it work previously without explicit dependency?

**Review by relu91 — APPROVED** [review]
Looks good just a couple of comments below:

File: packages/binding-modbus/src/modbus-client.ts (hunk 2 above)
Comment (fillobotto) [reply packages/binding-modbus/src/modbus-client.ts]: There was no usage of Subscription class inside modbus package previously so I just proceeded in the same was as other packages, for example https://github.com/relu91/thingweb.node-wot/blob/ceab936ff65e82b15498c024e7d2b923150c6c6f/packages/binding-coap/src/coap-client.ts#L23

**Review by relu91 — CHANGES_REQUESTED** [review]
I merged the current changes in node-wot into this branch. Now there are some falling test due to timeout can you fix them?

**Review by relu91 — APPROVED** [review]
looks good!

**Pull request merged**